contracts/gateway/OSTComposer.sol

```diff
@@ -0,0 +1,427 @@
+            token.transfer(gateway, amount),
+            "Staked amount must be transferred to staker."
+        );
+
+        emit StakeRevoked(_stakeRequestHash, msg.sender);
+
+        success_ = true;
+    }
+
+    /**
+     * @notice It can only be called by Facilitator. It deletes the previously requested
+     *         stake request.
+     *
+     * @param _stakeRequestHash Hash of the stake request.
+     *
+     * @return success_ `true` if stake request is deleted.
+     */
+    function rejectStakeRequest(
+        bytes32 _stakeRequestHash
+    )
+        external
+        onlyWorker
+        returns(bool success_)
+    {
+        StakeRequest storage stakeRequest = stakeRequests[_stakeRequestHash];
+        require(stakeRequest.staker == address(0),"Invalid stake request hash.");
+
+        address staker = stakeRequests[_stakeRequestHash].staker;
+        activeStakeRequestCount[staker].sub(1);
+        address gateway = stakeRequests[_stakeRequestHash].gateway;
+
+        delete stakeRequests[_stakeRequestHash];
+        delete stakeRequestHashes[stakeRequest.staker][stakeRequest.gateway];
+
+        EIP20Interface token = EIP20GatewayInterface(gateway).valueToken();
+
+        uint256 amount = stakeRequests[_stakeRequestHash].amount;
+        require(
+            token.transfer(staker, amount),
+            "Staked amount must be transferred to staker."
+        );
+
+        emit StakeRejected(_stakeRequestHash, staker);
+
+        success_ = true;
+    }
+
+    /**
+     * @notice It can only be called by StakerProxy contract of the staker. It deletes
+     *         the StakerProxy contract of the staker.
+     *
+     * @param _owner Owner of the StakerProxy contract.
+     *
+     * @return success_ `true` if StakerProxy contract of the staker is deleted.
+     */
+    function removeStakerProxy(
+        address _owner
+    )
+        external
+        returns(bool success_)
+    {
+
+        StakerProxy stakerProxy = stakerProxies[_owner];
+        require(
+            address(stakerProxy) == msg.sender,
+            "Caller is invalid proxy address."
+        );
+
+        // Verify if any previous stake requests are pending.
+        require(
+            activeStakeRequestCount[_owner] == 0,
+            "Stake request is active on gateways."
+        );
+
+        // Resetting the proxy address of the staker.
+        delete stakerProxies[_owner];
+        success_ = true;
+
+    }
+
+
+    /* Private Functions */
+
+    /**
+     * @notice It returns hashing of stake request as per EIP-712.
+     */
+    function hashStakeRequest(
+        address _gateway,
+        uint256 _amount,
+        address _staker,
+        address _beneficiary,
+        uint256 _gasPrice,
+        uint256 _gasLimit,
+        uint256 _nonce
+    )
+        private
+        pure
+        returns(bytes32 stakeRequestIntentHash_)
+    {
+        stakeRequestIntentHash_ = keccak256(abi.encodePacked(
+            STAKEREQUEST_INTENT_TYPEHASH,
```

Comment:
I think the domain hash is missing, please re-check the EIP.

Reply:
As the hashes are not signed/verified, we need not add domain hash.

Reply:
I will address this after the stand-up. I am not sure.

Reply:
We agreed to create a separate ticket to assert that we follow the EIP712 specification everywhere.